server/aesdsocket.c

```diff
@@ -0,0 +1,289 @@
+		syslog(LOG_INFO, "Running aesd socket as daemon\r\n");
+	}
+
+	memset(&hints, 0, sizeof(hints));
+	hints.ai_family = DOMAIN;
+	hints.ai_socktype = TYPE;
+	hints.ai_flags = AI_PASSIVE;
+
+	int err = getaddrinfo(NULL, PORT, &hints, &servinfo);
+
+	if (err != 0) {
+		syslog(LOG_ERR, "getaddrinfo() failed with an error: %s\r\n", strerror(errno));
+		freeaddrinfo(servinfo);
+		exit(EXIT_FAILURE);
+	}
+
+	if(servinfo == NULL)
+	{
+		syslog(LOG_ERR, "servinfo is NULL\r\n");
+		exit(EXIT_FAILURE);
+	}
+
+	sock_fd = socket(servinfo->ai_family, servinfo->ai_socktype, PROTOCOL);
+	if (sock_fd == -1)
+	{
+		syslog(LOG_ERR, "socket() failed with an error: %s\r\n", strerror(errno));
+		freeaddrinfo(servinfo);
+		exit(EXIT_FAILURE);
+	}
+
+	syslog(LOG_INFO, "Socket created successfully with fd: %d\r\n", sock_fd);
+
+	// Running the application numtiple times without the reuse code below results into: bind() failed with an error: Address already in use#015
+	if (setsockopt(sock_fd, SOL_SOCKET, SO_REUSEADDR, &yes, sizeof(int)) == -1) {
+		syslog(LOG_ERR, "setsockopt() failed with an error: %s\r\n", strerror(errno));
+		freeaddrinfo(servinfo);
+		exit(EXIT_FAILURE);
+	}	
+
+	if (bind(sock_fd, servinfo->ai_addr, sizeof(struct addrinfo)) == -1) {
+		syslog(LOG_ERR, "bind() failed with an error: %s\r\n", strerror(errno));
+		freeaddrinfo(servinfo);
+		exit(EXIT_FAILURE);
+	}
+
+	syslog(LOG_INFO, "Socket bound successfully \r\n");
+
+	if (servinfo != NULL)
+	{
+		freeaddrinfo(servinfo);
+		syslog(LOG_INFO, "servinfo freed \r\n");
+	}
+
+	if (run_as_daemon == true)
+	{
+		create_daemon();
+	}
+
+	if (listen(sock_fd, BACKLOG) == -1) {
+		syslog(LOG_ERR, "listen() failed with an error: %s\r\n", strerror(errno));
+		exit(EXIT_FAILURE);
+	}
+
+	bool packet_complete = false;
+	recv_buf = (char *)malloc(BUF_SIZE * sizeof(char));
+	int recv_bytes, bytes_written, bytes_read, bytes_sent = 0;
+
+	while(1) {  // main accept() loop
+		struct sockaddr_in client_addr;
+		socklen_t client_addr_len = sizeof(client_addr);
+		accept_fd = accept(sock_fd, (struct sockaddr *)&client_addr, &client_addr_len);
+		if (accept_fd == -1) {
+			syslog(LOG_ERR, "accept() failed with an error: %s\r\n", strerror(errno));
+			continue;
+		}
+
+		char client_ip[INET_ADDRSTRLEN];
+		inet_ntop(PF_INET, &(client_addr.sin_addr), client_ip, INET_ADDRSTRLEN );
+		syslog(LOG_INFO, "Accepted connection from %s\r\n", client_ip);
+
+		fd = open(FILE_NAME, O_CREAT | O_RDWR | O_APPEND, S_IWUSR | S_IRUSR | S_IRGRP | S_IROTH);
+		if(fd == -1) {
+			syslog(LOG_ERR, "Failed opening file %s with an error: %s", FILE_NAME, strerror(errno));
+			exit(EXIT_FAILURE);
+		}
+		syslog(LOG_INFO, "File %s opened for dumping the packet data\r\n", FILE_NAME);
+
+		do
+		{
+			recv_bytes = recv(accept_fd, recv_buf, BUF_SIZE, 0);
+			if (recv_bytes == -1)
+			{
+				syslog(LOG_ERR, "recv() failed with an error: %s\r\n", strerror(errno));
+				exit(EXIT_FAILURE);
+			}
+
+			bytes_written = write(fd, recv_buf, recv_bytes);
```

Comment:
@krishnasuhagiya29 with this design you will write to the file before the packet completes, which may cause issues with future assignments.